src/opencensus/proto/metrics/v1/metrics.proto

```diff
@@ -195,26 +197,37 @@ message DistributionValue {
 
   // A Distribution may optionally contain a histogram of the values in the
   // population. The bucket boundaries for that histogram are described by
-  // bucket_bounds. This defines size(bucket_bounds) + 1 (= N)
-  // buckets. The boundaries for bucket index i are:
-  //
-  // (-infinity, bucket_bounds[i]) for i == 0
-  // [bucket_bounds[i-1], bucket_bounds[i]) for 0 < i < N-2
-  // [bucket_bounds[i-1], +infinity) for i == N-1
+  // BucketOptions.
   //
-  // i.e. an underflow bucket (number 0), zero or more finite buckets (1
-  // through N - 2, and an overflow bucket (N - 1), with inclusive lower
-  // bounds and exclusive upper bounds.
-  //
-  // If bucket_bounds has no elements (zero size), then there is no
-  // histogram associated with the Distribution. If bucket_bounds has only
-  // one element, there are no finite buckets, and that single element is the
-  // common boundary of the overflow and underflow buckets. The values must
-  // be monotonically increasing.
-  //
-  // Don't change bucket boundaries within a timeseries if your backend
-  // doesn't support this.
-  repeated double bucket_bounds = 4;
+  // If bucket_options has no type, then there is no histogram associated with
+  // the Distribution.
+  message BucketOptions {
+    oneof type {
+      // Bucket with explicit bounds.
+      Explicit explict = 1;
+    }
+
+    // Specifies a set of buckets with arbitrary upper-bounds.
+    // This defines size(bounds) + 1 (= N) buckets. The boundaries for bucket
+    // index i are:
+    //
+    // [0, bucket_bounds[i]) for i == 0
```

Comment:
So the histogram will only accommodate non-negative values? Is this compatible with our current APIs?

Reply:
Our current APIs (in stats) do support negative values but no back-end that I investigated (Prometheus or Stackdriver) support this. So we will need to make this change in stats to only record positive values.

Reply:
SGTM. In this case we need to update the gRPC metrics (https://github.com/census-instrumentation/opencensus-specs/blob/master/stats/gRPC.md#units) as well.

Reply:
@bogdandrutu what happens to users that would like to record temperature values and would like to OpenCensus to collect values then trigger alerts such as warming up the house? We published a blogpost such as https://opencensus.io/articles/iot/ and anyone in Canada (where I also hail from and have witnessed -40C=-40F many times and most of the year temperatures are below 0C) who might repeat that experiment might be surprised after this change. Please correct me if am wrong

Reply:
@odeke-em I will give you two answers:
1. Funny answer: they can use K (Kelvin) and it is guaranteed to be greater than 0 (or if it gets to 0 it will be too late for this to matter).
2. Serious answer: currently no backend supports negative values in a distribution (Prometheus no, Stackdriver says it supports negative values, but I tested and it does not work, SignalFx not yet supports histograms). And they have good reason to not support this.

So yes that blog post can be misleading.

Reply:
Thanks for the responses @bogdandrutu, gotcha. We'll perhaps also need to add a note in the specs repo that bucket bounds begin at 0.